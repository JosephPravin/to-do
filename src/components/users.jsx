import React, { Component } from 'react';
import User from './user';
import AddUser from './addUser';

class Users extends Component {

    render() { 
        const { users } = this.props;
        return ( 
        <div className="container">
            <div className="row">
                {
                    users.map(user => (
                        <User className="col-sm" name={user.name} tasks={user.tasks} onAddTask={this.props.onAddTask}></User>
                    ))
                }  
            </div>
        </div>
         );
    }
}
 
export default Users;