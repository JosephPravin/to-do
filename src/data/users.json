[
    {
    "id":"1",
    "name": "Joseph",
    "tasks": [
        {
            "title": "Button link inactive",
            "description": "Though home page button should be active, it is inactive on page load"
        }
    ]
    },
    {
    "id":"2",
    "name": "Pravin",
    "tasks": [
            {
                "title": "Jenkins task",
                "description": "Create a jenkins job for UI deployment in dev. Branch name should be parameterised"
            },{
                "title": "Text overflowing in title card",
                "description": "Css changes to accomodate the text within the div space"
            }
        ]
    },
    {
    "id":"3",
    "name": "Rajaram",
    "tasks": []
    }
]