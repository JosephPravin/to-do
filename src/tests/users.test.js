import React from 'react';
import renderer from "react-test-renderer";
import Users from '../components/users';
import users from '../data/users.json';

test("App snapshot", () => {
  const component = renderer.create(<Users users={users} onAddTask={()=>{}} onDeleteTask={()=>{}}/>);
  expect(component.toJSON()).toMatchSnapshot();
});
