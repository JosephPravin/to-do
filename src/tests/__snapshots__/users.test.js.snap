// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Joseph
        </mark>
      </h3>
      <div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Button link inactive
            </h5>
            <p
              className="card-text bg-light"
            >
              Though home page button should be active, it is inactive on page load
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
      </div>
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Pravin
        </mark>
      </h3>
      <div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Jenkins task
            </h5>
            <p
              className="card-text bg-light"
            >
              Create a jenkins job for UI deployment in dev. Branch name should be parameterised
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Text overflowing in title card
            </h5>
            <p
              className="card-text bg-light"
            >
              Css changes to accomodate the text within the div space
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
      </div>
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Rajaram
        </mark>
      </h3>
      <div />
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshots App snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Joseph
        </mark>
      </h3>
      <div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Button link inactive
            </h5>
            <p
              className="card-text bg-light"
            >
              Though home page button should be active, it is inactive on page load
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
      </div>
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Pravin
        </mark>
      </h3>
      <div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Jenkins task
            </h5>
            <p
              className="card-text bg-light"
            >
              Create a jenkins job for UI deployment in dev. Branch name should be parameterised
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Text overflowing in title card
            </h5>
            <p
              className="card-text bg-light"
            >
              Css changes to accomodate the text within the div space
            </p>
            <a
              class="card-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "red",
                  "float": "right",
                }
              }
            >
              Delete
            </a>
          </div>
        </div>
      </div>
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
    <div
      className="col"
      style={
        Object {
          "padding": "1em",
        }
      }
    >
      <h3>
        <mark>
          Rajaram
        </mark>
      </h3>
      <div />
      <button
        className="btn btn-sm btn-dark"
        onClick={[Function]}
      >
        Add task
      </button>
    </div>
  </div>
</div>
`;
